src/server.c

```diff
@@ -3644,17 +3644,16 @@ int processCommand(client *c) {
         !(!(c->cmd->flags&CMD_MOVABLE_KEYS) && c->cmd->key_specs_num == 0 &&
           c->cmd->proc != execCommand))
     {
-        int hashslot;
         int error_code;
         clusterNode *n = getNodeByQuery(c,c->cmd,c->argv,c->argc,
-                                        &hashslot,&error_code);
+                                        &c->slot,&error_code);
```

Comment:
Was this resolved based on your later comment?

Reply:
yes, it can be folded into the other discussion. if we conclude we don't wanna mix declared keys with actual ones, we should move the var, otherwise we must keep it like it is.